nfc/NfcCx/Cx/NfcCxRF.cpp

```diff
@@ -141,6 +141,33 @@ NfcCxRFInterfaceGetSEEvent(
 }
 
 VOID FORCEINLINE
+NfcCxRFInterfaceClearTagRemoteDevList(
+    _Inout_ PNFCCX_RF_INTERFACE RFInterface
+)
+{
+    uint8_t bIndex = 0x00;
+
+    if (RFInterface->pLibNfcContext->pRemDevList != NULL) 
+    {
+        for (bIndex = 0; bIndex < RFInterface->pLibNfcContext->uNoRemoteDevices; bIndex++)
+        {
+            if (RFInterface->pLibNfcContext->pRemDevList[bIndex].psRemoteDevInfo != NULL)
+            {
+                if ((RFInterface->pLibNfcContext->pRemDevList[bIndex].psRemoteDevInfo->RemDevType == phLibNfc_eISO14443_A_PCD) ||
+                    (RFInterface->pLibNfcContext->pRemDevList[bIndex].psRemoteDevInfo->RemDevType == phLibNfc_eISO14443_B_PCD))
+                {
+                    RFInterface->pLibNfcContext->bIsHCEConnected = FALSE;
+                }
+                free(RFInterface->pLibNfcContext->pRemDevList[bIndex].psRemoteDevInfo);
+                RFInterface->pLibNfcContext->pRemDevList[bIndex].psRemoteDevInfo = NULL;
+            }
+        }
+        free(RFInterface->pLibNfcContext->pRemDevList);
+        RFInterface->pLibNfcContext->pRemDevList = NULL;
+    }
+}
+
```

Comment:
Addressing this comment requires changes in legacy CLX code as well.
As discussed, we would take up these changes separately ( post RS2 feature freeze).